ubuntu_from_source/Dockerfile

```
@@ -0,0 +1,46 @@
+ARG UBUNTU_VERSION=20.04
+
+FROM ubuntu:$UBUNTU_VERSION
+
+LABEL maintainer.name="ROOT team"
+LABEL maintainer.email="root-dev@cern.ch"
+
+ENV LANG=C.UTF-8
+
+COPY packages packages
+
+RUN apt-get update -qq && \
+    ln -sf /usr/share/zoneinfo/UTC /etc/localtime && \
+    apt-get -y install $(cat packages) wget git && \
+    apt-get autoremove -y && \
+    apt-get clean -y && \
+    rm -rf /var/cache/apt/archives/* && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV ROOTSYS /opt/root
+ENV PATH $ROOTSYS/bin:$PATH
+ENV PYTHONPATH $ROOTSYS/lib:$PYTHONPATH
+ENV CLING_STANDARD_PCH none
+
+RUN echo $ROOTSYS/lib >> /etc/ld.so.conf && \
+    ldconfig
+
+RUN yes | unminimize
+
+ARG CMAKE_CXX_STANDARD=17
+ARG ROOT_VERSION=master
+
+# clone build and install
+RUN git clone https://github.com/root-project/root.git $ROOTSYS/source \
+    && cd $ROOTSYS/source && git reset --hard ${ROOT_VERSION} \

```

No need to clone the whole repo:

```suggestion
RUN git clone --branch ${GIT_REVISION} --depth=1 https://github.com/root-project/root.git $ROOTSYS/source \
    && cd $ROOTSYS/source \

```

---

This was my original way of getting the source code, but this doesn't work with commits if I remember correctly. If we only want git branches / tags this can be changed to what you suggest.

---

In 99% of cases we'll build a tag or the tip of a branch, so I think the bandwidth savings are worth it.

in the rare case anyone wants to build a specific commit using this dockerfile, they should know how to edit the git clone command.